packages/vue3/src/_dev/App.vue

```diff
@@ -1,26 +1,36 @@
 <script setup lang="ts">
 import 'uno.css';
 import '../assets/main.css';
-import { ref } from 'vue';
-import { PrimaryButton, Toast } from '~/components';
+import { ref, watch } from 'vue';
+import { MediaInput, UploadedMedia } from '~/components';
 
-const show = ref(false);
+const attachments = ref<File[]>([]);
+const disabled = ref(false);
+const limit = ref(4);
+
+const checkLimit = () => {
+  disabled.value = attachments.value.length >= limit.value;
+};
+
+const deleteFile = (file: File) => {
```

Comment:
Perhaps a better way of managing deletion is this:
```suggestion
const deleteFile = (file: File) => {
  attachments.value = attachments.value.filter(f => f.name !== file.name);
};
```